src/test/java/core/basesyntax/service/ReaderServiceImplTest.java

```diff
@@ -0,0 +1,38 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import service.ReaderService;
+import service.impl.ReaderServiceImpl;
+
+public class ReaderServiceImplTest {
+    private static final String VALID_INPUT_FILE_PATH = "src/test/resources/input.csv";
+    private static final List<String> INPUT_FILE_DATA = List.of(
+            "b,banana,20",
+            "b,grape,50",
+            "b,apple,100",
+            "s,banana,100",
+            "p,banana,13",
+            "r,apple,10",
+            "p,apple,20",
+            "p,banana,5",
+            "s,banana,50",
+            "p,grape,30"
+    );
```

Comment:
instantiate expected\test data inside test case. NO NEED TO MAKE EVERYTHING CONSTANTS.
```suggestion
    private static final String VALID_INPUT_FILE_PATH = "src/test/resources/input.csv";
    private static final List<String> INPUT_FILE_DATA = List.of(
            "b,banana,20",
            "b,grape,50",
            "b,apple,100",
            "s,banana,100",
            "p,banana,13",
            "r,apple,10",
            "p,apple,20",
            "p,banana,5",
            "s,banana,50",
            "p,grape,30"
    );
```